dev/MRTCore/mrt/Core/src/MRM.cpp

```diff
@@ -874,49 +876,20 @@ STDAPI MrmGetFilePathFromName(_In_opt_ PCWSTR filename, _Outptr_ PWSTR* filePath
 {
     *filePath = nullptr;
 
-    wchar_t path[MAX_PATH];
-    DWORD size = ARRAYSIZE(path);
-    DWORD length = GetModuleFileName(nullptr, path, size);
-    DWORD lastError = GetLastError();
-    std::unique_ptr<wchar_t[]> pathAllocated;
+    wil::unique_cotaskmem_string path;
+    RETURN_IF_FAILED((wil::GetModuleFileNameW<wil::unique_cotaskmem_string, 256>(nullptr, path)));
 
-    while ((length == size) && (lastError == ERROR_INSUFFICIENT_BUFFER))
-    {
-        size *= 2;
-        RETURN_HR_IF(E_UNEXPECTED, size > 32 * 1024);
-
-        pathAllocated.reset(new (std::nothrow) wchar_t[size]);
-        RETURN_IF_NULL_ALLOC(pathAllocated);
-
-        length = GetModuleFileName(nullptr, pathAllocated.get(), size);
-        lastError = GetLastError();
-    }
-
-    RETURN_HR_IF(HRESULT_FROM_WIN32(lastError), lastError != 0);
-
-    // Remove module file name
-    PWSTR pointerToPath = pathAllocated ? pathAllocated.get() : path;
-    PWSTR lastSlash = nullptr;
-    PWSTR secondToLastSlash = nullptr;
-    while (*pointerToPath)
-    {
-        if (*pointerToPath == L'\\')
-        {
-            secondToLastSlash = lastSlash;
-            lastSlash = pointerToPath;
-        }
-        pointerToPath++;
-    }
+    PCWSTR name = wil::find_last_path_segment(path.get());
+    wchar_t moduleFilename[MAX_PATH];
```

Comment:
883-884 can go into the "// Prep for the second iteration" if block where if name is empty/null, we just break. We can avoid the unnecessary setting of module name to path and then a second iteration.

Reply:
at that time, path is already modified, and lost the name.